src/docs/testing/unit-testing.md

```diff
@@ -29,14 +29,34 @@ import { MyCmp } from '../my-cmp';
 it('should render my component', async () => {
   const page = await newSpecPage({
     components: [MyCmp],
-    html: `<my-cmp></my-cmp>`,
+    html: `<my-cmp title="Hello World"></my-cmp>`,
   });
   expect(page.root).toEqualHtml(`
-    <my-cmp>Success!</my-cmp>
+    <my-cmp>Hello World</my-cmp>
   `);
 });
 ```
 
+The example below uses the template option to test the component
+```
+// mycmp.spec.tsx
+// This is using jsx syntax and should be in a tsx file.
+import { h } from '@stencil/core';
+import { newSpecPage } from '@stencil/core/testing';
+import { MyCmp } from '../my-cmp';
+
+it('should render my component', async () => {
+  const title = 'Hello World';
+  const page = await newSpecPage({
+    components: [MyCmp],
+    template: () => (<my-cmp title={title}></my-cmp>),
```

Comment:
If someone were to use `title` as a `@Prop`, they'd get a warning like:

>  [ WARN  ]  Build Warn: ./src/components/my-cmp.tsx:12:11
           The @Prop() name "title" is a reserved public name. Please rename the
           "title" prop so it does not conflict with an existing standardized prototype
           member. Reusing prop names that are already defined on the element's
           prototype may cause unexpected runtime errors or user-interface issues on
           various browsers, so it's best to avoid them entirely.

It may be helpful to folks that aren't used to this warning to rename the attribute here something like 'greeting' or 'message':
```suggestion
  const greeting = 'Hello World';
  const page = await newSpecPage({
    components: [MyCmp],
    template: () => (<my-cmp greeting={greeting}></my-cmp>),
```